### PR TITLE
Upgrade to statemachine 1.1.0

### LINKED ISSUE #8: Upgrade to StateMachine's latest GA release
As a developer, I'd like to upgrade the deployer code to adapt to StateMachine's upcoming GA release.

### PR DESCRIPTION
**Pull request by jvalkeal**
NOTE: goes together with https://github.com/spring-cloud/spring-cloud-dataflow-server-yarn/pull/106
- Upgrade to statemachine release version.
- Moving to uml model for machine as it is
  easier to design and understand.
- Adding internal error handling logic in a
  state handling.
- Overhauled a lot of machine state handling.
- Fixes #8

### CODE REVIEW

**Comment by jvalkeal** [comment]
Logic for machines are per these statecharts:

appdeployer-model

tasklauncher-model

File: spring-cloud-deployer-yarn-build-tests/src/test/java/org/springframework/cloud/dataflow/yarn/buildtests/AppDeployerIT.java [hunk 1]
```diff
@@ -486,10 +487,21 @@ public TaskExecutor appDeployerTaskExecutor() {
 			return executor;
 		}
 
+//		@Bean
```
Comment (trisberg) [inline spring-cloud-deployer-yarn-build-tests/src/test/java/org/springframework/cloud/dataflow/yarn/buildtests/AppDeployerIT.java]: Why add commented out code?

**Comment by trisberg** [comment]
Get a build error on AppDeployerIT:

```
$ more /home/vagrant/spring-cloud-deployer-yarn/spring-cloud-deployer-yarn-build-tests/target/failsafe-reports/org.springframework.cloud.dataflow.yarn.buildtests.AppDeployerIT.txt 
-------------------------------------------------------------------------------
Test set: org.springframework.cloud.dataflow.yarn.buildtests.AppDeployerIT
-------------------------------------------------------------------------------
Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 363.325 sec <<< FAILURE! - in org.springframework.cloud.dataflow.yarn.buildtests.AppDeployerIT
testSmokeDeployer(org.springframework.cloud.dataflow.yarn.buildtests.AppDeployerIT)  Time elapsed: 0.643 sec  <<< ERROR!
java.lang.NullPointerException: null
    at org.springframework.statemachine.support.AbstractStateMachine.acceptEvent(AbstractStateMachine.java:690)
    at org.springframework.statemachine.support.AbstractStateMachine.sendEvent(AbstractStateMachine.java:217)
    at org.springframework.cloud.deployer.spi.yarn.YarnAppDeployer.deploy(YarnAppDeployer.java:162)
    at org.springframework.cloud.dataflow.yarn.buildtests.AppDeployerIT.testSmokeDeployer(AppDeployerIT.java:395)
```

File: spring-cloud-deployer-yarn-build-tests/src/test/java/org/springframework/cloud/dataflow/yarn/buildtests/AppDeployerIT.java (hunk 1 above)
Comment (jvalkeal) [inline spring-cloud-deployer-yarn-build-tests/src/test/java/org/springframework/cloud/dataflow/yarn/buildtests/AppDeployerIT.java]: yeah, my bad.

**Comment by trisberg** [comment]
Finally got tests to pass after allocating all of my CPUs to the VM I used! Will merge soon ...

**Pull request merged**